juju/juju
# [JUJU-2000] Fix deploy revision refresh test regression

A few things were lost in merges from 2.9 into develop or 3.0, bring them back.
* loop to ensure that deploy by revision refresh is working with async charm download, from #14368
* latest vs default track, from #14470

## QA steps

*Commands to run to verify that the change works.*

From #14470:
```sh
$ juju deploy mysql-router
Located charm "mysql-router" in charm-hub, revision 34
Deploying "mysql-router" from charm-hub charm "mysql-router", revision 34 in channel 8.0/stable on focal
$ juju deploy mysql-router --channel latest mysql-router-latest
Located charm "mysql-router" in charm-hub, revision 15
Deploying "mysql-router-latest" from charm-hub charm "mysql-router", revision 15 in channel latest/stable on focal

# the channel here should be latest/stable, however there is a bug in charmhub server right now.
$ juju deploy ubuntu
Located charm "ubuntu" in charm-hub, revision 20
Deploying "ubuntu" from charm-hub charm "ubuntu", revision 20 in channel stable on focal
```

Other QA
```sh
(cd tests ; ./main.sh -v deploy test_deploy_revision)
(cd tests ; ./main.sh -v deploy test_deploy_bundles)
```
